cmd/d2ctl/cmd/root.go

```diff
@@ -12,6 +12,20 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// StatePath stores the path to the local directory for Pulumi state.
+var StatePath string
+
+func init() {
+	// Use current user home directory as base for pulumi backend store
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+
+	RootCmd.PersistentFlags().StringVar(&StatePath, "state-path", homeDir+"/.day-two-state", "Path to store the local state")
```

Comment:
Turbo-nit: we should use filepath.join instead of concatenating the path. Ensures portability from what I can tell from a quick google. Once fixed, the rest LGTM